src/mongo/db/commands/dbcommands.cpp

```diff
@@ -316,7 +316,7 @@ class CmdProfile : public Command {
                                        const BSONObj& cmdObj) {
         AuthorizationSession* authzSession = AuthorizationSession::get(client);
 
-        if (cmdObj.firstElement().numberInt() == -1 && !cmdObj.hasField("slowms")) {
+        if (cmdObj.firstElement().numberInt() == -1 && !cmdObj.hasField("slowms") && !cmdObj.hasField("samplerate")) {
```

Comment:
I'd like to see a test proving that if you include sampleRate in the profile command, then enableProfiler privileges are required. This is probably easiest to accomplish by adding a new test case to jstests/auth/lib/commands_lib.js. You can see examples of the other profile command authorization tests here:

[https://github.com/mongodb/mongo/blob/r3.5.1/jstests/auth/lib/commands_lib.js#L3515-L3573](url)